components/brave_news/browser/resources/Feed.tsx

```diff
@@ -121,33 +125,77 @@ export default function Component({ feed }: Props) {
 
     loadMoreObserver.current.observe(el);
   }, [])
+
+  // Create intersection observer & relevant state to measure
+  // the amount of viewed cards in the session.
+  const lastViewedCardCount = React.useRef(0);
+  const viewDepthIntersectionObserver = React.useRef(new IntersectionObserver(entries => {
+    const inViewCounts = entries
+      .filter(e => e.intersectionRatio === 1)
+      .map(e => Number(e.target.getAttribute(CARD_COUNT_ATTRIBUTE)));
+    if (!inViewCounts.length) {
+      return;
+    }
+    const largestCardCount = Math.max(...inViewCounts);
+    // Ensure we only report increases in scroll depth
+    // by comparing to the last scroll card count
+    if (lastViewedCardCount.current >= largestCardCount) {
+      return;
+    }
+    const newViews = largestCardCount - lastViewedCardCount.current;
+    if (!!onSessionStart && lastViewedCardCount.current === 0 && newViews > 0) {
```

Comment:
Optional: `!!` isn't needed
```suggestion
    if (onSessionStart && lastViewedCardCount.current === 0 && newViews > 0) {
```